### ISSUE TITLE
Node Version

### ISSUE THREAD

**Issue by rezocharly** [eouia/MMM-Assistant#10]
Hi !

I’m trying to setup the MMM-Assistant module. I followed the wiki install process but at the end I have an error in my console and I can’t launch the mirror.

Here is the error;

```
WARNING! Could not load config file. Starting with default configuration. 
Error found: Error: The module '/home/pi/MagicMirror/modules/MMM-Assistant/node_modules/speaker/build/Release/binding.node’
was compiled against a different Node.js version using
NODE_MODULE_VERSION 48. This version of Node.js requires
NODE_MODULE_VERSION 54. Please try re-compiling or re-installing
the module (for instance, using npm rebuild ornpm install).
```

I tried the npm rebuild, npm install, delete the node_modules folder and then npm install again, but I still have this error.

Thanks a lot for your help !

**Comment by eouia** [comment]
I think you might have missed step 4 in your installation.

**Comment by rezocharly** [comment]
Hi Eouia, 

I followed step by step. This error came after step 4.

**Comment by rezocharly** [comment]
I started again from the beginning. 

Now I have this error: 

App threw an error during load
Error: Failed to load gRPC binary module because it was not installed for the current system
Expected directory: electron-v1.7-linux-arm-glibc
Found: [electron-v1.7-linux-arm-{libc}, node-v48-linux-arm-glibc]
This problem can often be fixed by running "npm rebuild" on the current system
Original error: Cannot find module '/home/pi/MagicMirror/modules/MMM-Assistant/node_modules/grpc/src/node/extension_binary/electron-v1.7-linux-arm-glibc/grpc_node.node'
    at Object.<anonymous> (/home/pi/MagicMirror/modules/MMM-Assistant/node_modules/grpc/src/grpc_extension.js:44:17)
    at Object.<anonymous> (/home/pi/MagicMirror/modules/MMM-Assistant/node_modules/grpc/src/grpc_extension.js:54:3)
    at Module._compile (module.js:571:32)
    at Object.Module._extensions..js (module.js:580:10)
    at Module.load (module.js:488:32)
    at tryModuleLoad (module.js:447:12)
    at Function.Module._load (module.js:439:3)
    at Module.require (module.js:498:17)
    at require (internal/module.js:20:19)
    at Object.<anonymous> (/home/pi/MagicMirror/modules/MMM-Assistant/node_modules/grpc/src/client.js:38:12)
Whoops! There was an uncaught exception...

If I do the npm rebuild I come back to the previous error about node version.

**Comment by rezocharly** [comment]
I fixed this issue with :  `cd ~/MagicMirror/modules/MMM-Assistant 
$ npm rebuild grpc --build-from-source`

But now I'm facing a new issue:

App threw an error during load
Error: /home/pi/MagicMirror/modules/MMM-Assistant/node_modules/google-assistant/node_modules/grpc/src/node/extension_binary/grpc_node.node: undefined symbol: X509_NAME_free
    at process.module.(anonymous function) [as dlopen] (ELECTRON_ASAR.js:173:20)
    at Object.Module._extensions..node (module.js:598:18)
    at Object.module.(anonymous function) [as .node] (ELECTRON_ASAR.js:173:20)
    at Module.load (module.js:488:32)
    at tryModuleLoad (module.js:447:12)
    at Function.Module._load (module.js:439:3)
    at Module.require (module.js:498:17)
    at require (internal/module.js:20:19)
    at Object.<anonymous> (/home/pi/MagicMirror/modules/MMM-Assistant/node_modules/google-assistant/node_modules/grpc/src/node/src/grpc_extension.js:38:15)
    at Object.<anonymous> (/home/pi/MagicMirror/modules/MMM-Assistant/node_modules/google-assistant/node_modules/grpc/src/node/src/grpc_extension.js:42:3)
Whoops! There was an uncaught exception...

**Comment by eouia** [comment]
Sorry. I believe you'd missed something on installation, but I can not track all the possible errors. 
In most case, It could be related to the node version among the MagicMirror&Electron and dependency modules. 
This MMM-Assistant needs so many dependencies, and there might be some gap of the temporary version of node modules updated. I suggest you try to build this again after one or two days from scratch.

**Comment by normyx** [comment]
Hi,
I got exactly the same issue :+1: Expected directory: electron-v1.4-linux-arm-glibc
Found: [electron-v1.4-linux-arm-{libc}, node-v59-linux-arm-glibc]
This problem can often be fixed by running "npm rebuild" on the current system
Original error: Cannot find module '/home/pi/MagicMirror/modules/MMM-Assistant/node_modules/grpc/src/node/extension_binary/electron-v1.4-linux-arm-glibc/grpc_node.node'
    at Object.<anonymous> (/home/pi/MagicMirror/modules/MMM-Assistant/node_modules/grpc/src/grpc_extension.js:53:17)
    at Module._compile (module.js:556:32)
    at Object.Module._extensions..js (module.js:565:10)
    at Module.load (module.js:473:32)
    at tryModuleLoad (module.js:432:12)
    at Function.Module._load (module.js:424:3)
    at Module.require (module.js:483:17)
    at require (internal/module.js:20:19)
    at Object.<anonymous> (/home/pi/MagicMirror/modules/MMM-Assistant/node_modules/grpc/src/client.js:37:12)
    at Module._compile (module.js:556:32)
Whoops! There was an uncaught exception...
{ Error: Failed to load gRPC binary module because it was not installed for the current system
Expected directory: electron-v1.4-linux-arm-glibc
Found: [electron-v1.4-linux-arm-{libc}, node-v59-linux-arm-glibc]
This problem can often be fixed by running "npm rebuild" on the current system
Original error: Cannot find module '/home/pi/MagicMirror/modules/MMM-Assistant/node_modules/grpc/src/node/extension_binary/electron-v1.4-linux-arm-glibc/grpc_node.node'
    at Object.<anonymous> (/home/pi/MagicMirror/modules/MMM-Assistant/node_modules/grpc/src/grpc_extension.js:53:17)
    at Module._compile (module.js:556:32)
    at Object.Module._extensions..js (module.js:565:10)
    at Module.load (module.js:473:32)
    at tryModuleLoad (module.js:432:12)
    at Function.Module._load (module.js:424:3)
    at Module.require (module.js:483:17)
    at require (internal/module.js:20:19)
    at Object.<anonymous> (/home/pi/MagicMirror/modules/MMM-Assistant/node_modules/grpc/src/client.js:37:12)
    at Module._compile (module.js:556:32) code: 'MODULE_NOT_FOUND' }
MagicMirror will not quit, but it might be a good idea to check why this happened. Maybe no internet connection?
If you think this really is an issue, please open an issue on GitHub: https://github.com/MichMich/MagicMirror/issues
Launching application.


DO YOU HAVE ANY CLUE (I WILL TRY IT FROM A SCRATCH RASPBIAN BUT NOT SURE IT WILL FIX)

**Comment by eouia** [comment]
@normyx 
rename `electron-v1.4-linux-arm-{libc}` to `electron-v1.4-linux-arm-glibc`.

**Comment by normyx** [comment]
Hi, thx, i will try this.

**Comment by E3V3A** [comment]
@normyx 
I assume this worked because of your recent PR.
